app/src/test/java/nu/yona/app/api/manager/impl/AuthenticateManagerImplTest.java

```diff
@@ -10,65 +10,230 @@
+import org.mockito.stubbing.Answer;
+import org.robolectric.RuntimeEnvironment;
 
 import nu.yona.app.YonaApplication;
+import nu.yona.app.YonaTestCase;
 import nu.yona.app.api.manager.AuthenticateManager;
+import nu.yona.app.api.manager.dao.AuthenticateDAO;
+import nu.yona.app.api.manager.network.AuthenticateNetworkImpl;
 import nu.yona.app.api.model.ErrorMessage;
+import nu.yona.app.api.model.Href;
+import nu.yona.app.api.model.Links;
+import nu.yona.app.api.model.OTPVerficationCode;
 import nu.yona.app.api.model.RegisterUser;
 import nu.yona.app.api.model.User;
 import nu.yona.app.listener.DataLoadListener;
+import nu.yona.app.listener.DataLoadListenerImpl;
 
+import static com.google.android.gms.internal.zzs.TAG;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
+
 /**
  * Created by kinnarvasa on 02/04/16.
  */
-public class AuthenticateManagerImplTest {
-
-    private final String password = "12423423234234324234";
+public class AuthenticateManagerImplTest extends YonaTestCase {
     private AuthenticateManagerImpl manager;
+    private AuthenticateNetworkImpl authenticateNetworkImplMock;
+    private AuthenticateDAO authenticateDAOMock;
+    private RegisterUser registerUser;
+
+    private String correctMobileNumber = "+919686270640";
+    private String wrongMobileNumber = "+9999211";
+    private OTPVerficationCode correctOtpCode = new OTPVerficationCode("1234");
+    private OTPVerficationCode wrongOtpCode = new OTPVerficationCode("1111");
+    private String nullOtpCode = null;
+
+    private DataLoadListener genericResponseListener = new DataLoadListener() {
+        @Override
+        public void onDataLoad(Object result) {
+            assertTrue(result instanceof User);
+        }
+        @Override
+        public void onError(Object errorMessage) {
+            assertTrue(errorMessage instanceof  ErrorMessage);
+        }
+    };
+
 
     @Before
     public void setUp() throws Exception {
+        setUpApplicationTestData();
+        setUpRegisterUser();
         manager = new AuthenticateManagerImpl(YonaApplication.getAppContext());
+        mockRequiredClasses();
+        setUpMockedAuthNetworkDaoMethods();
+        setUpMockedAuthNetworkImplMethods();
     }
 
     @Test
-    public void checkRegisterUser() {
-        RegisterUser registerUser = new RegisterUser();
-        registerUser.setFirstName("Kinnar");
-        registerUser.setLastName("Vasa");
-        registerUser.setMobileNumber("+31873449748");
-        registerUser.setNickName("Kinnar");
-
-        manager.registerUser(registerUser, true, new DataLoadListener() {
-            @Override
-            public void onDataLoad(Object result) {
-                validateMobileNumber(password, ((User) result).getMobileNumberConfirmationCode());
-                assertTrue(result instanceof User);
-            }
+    public void verifyUserRegistrationWithProperData() {
+        registerUser.setMobileNumber(correctMobileNumber);
```

Comment:
Why do these tests not have any asserts?

Reply:
The asserts are written in ```DataloadListener``` class callbacks .

Reply:
These are insufficient.
Take this example: ``verifyUserPassCodeWithWrongCode``. That test needs to assert that the data load listener was called with an error. All tests need similar asserts to verify correct behavior.